R/with.R

```diff
@@ -31,8 +31,11 @@
 #' names(my_undesirable_functions)
 #' @export
 modify_defaults <- function(defaults, ...) {
-  if (missing(defaults) || !is.list(defaults) || !all(nzchar(names2(defaults)))) {
-    cli_abort("{.arg defaults} must be a named list.")
+  if (missing(defaults)) {
+    cli_abort("Missing required argument {.arg defaults} should be a named list.")
```

Comment:
I think this error message should only mention that the required argument is missing. 
"Missing required argument" sounds a bit odd, and combines the error message for two different checks into one.

```r
cli_abort("{.arg defaults} is a required argument, but is missing. [It should be a named list.]") # second sentence is optional
```

Reply:
I agree, it would mimic more what `rlang::check_required()` provides. I can make this change. and adjust test.

```r
! `x` is absent but must be supplied.
```